### ISSUE TITLE
Attempting to open a file picker on macOS causes an exception.

### ISSUE THREAD

**Issue by vulppine** [space-wizards/RobustToolbox#2612]
This might be a threading issue related to GLFW, I've attached a report below.

<details>
  <summary>Log</summary>

```
Thread 0:: Dispatch queue: com.apple.main-thread
0   com.apple.CoreFoundation      	0x00007fff203ee008 CFAllocatorAllocate + 57
1   com.apple.CoreFoundation      	0x00007fff2046955e CFRunLoopRunSpecific + 373
2   com.apple.HIToolbox           	0x00007fff286afa83 RunCurrentEventLoopInMode + 292
3   com.apple.HIToolbox           	0x00007fff286af6b6 ReceiveNextEventCommon + 284
4   com.apple.HIToolbox           	0x00007fff286af583 _BlockUntilNextEventMatchingListInModeWithFilter + 70
5   com.apple.AppKit              	0x00007fff22c71172 _DPSNextEvent + 864
6   com.apple.AppKit              	0x00007fff22c6f945 -[NSApplication(NSEvent) _nextEventMatchingEventMask:untilDate:inMode:dequeue:] + 1364
7   libglfw.3.dylib               	0x00000001103011b2 _glfwPlatformWaitEventsTimeout + 306
8   ???                           	0x0000000113d40ec0 0 + 4627631808
9   ???                           	0x0000000113d40e30 0 + 4627631664
10  ???                           	0x0000000113d40ca3 0 + 4627631267
11  ???                           	0x0000000113d40aa4 0 + 4627630756
12  ???                           	0x0000000113133178 0 + 4614992248
13  ???                           	0x00000001124a6a1e 0 + 4601834014
14  ???                           	0x00000001124a4da2 0 + 4601826722
15  ???                           	0x00000001124a3eb9 0 + 4601822905
16  ???                           	0x00000001124a3cdc 0 + 4601822428
17  libcoreclr.dylib              	0x000000010b867819 CallDescrWorkerInternal + 124
18  libcoreclr.dylib              	0x000000010b6b2928 MethodDescCallSite::CallTargetWorker(unsigned long const*, unsigned long*, int) + 1496
...
511 libcoreclr.dylib              	0x000000010b6b2928 MethodDescCallSite::CallTargetWorker(unsigned long const*, unsigned long*, int) + 1496

...

Thread 15 Crashed:
0   libsystem_kernel.dylib        	0x00007fff2034392e __pthread_kill + 10
1   libsystem_pthread.dylib       	0x00007fff203725bd pthread_kill + 263
2   libsystem_c.dylib             	0x00007fff202c74a0 __abort + 139
3   libsystem_c.dylib             	0x00007fff202c7415 abort + 140
4   libc++abi.dylib               	0x00007fff20335ef2 abort_message + 241
5   libc++abi.dylib               	0x00007fff203275fd demangling_terminate_handler() + 266
6   libobjc.A.dylib               	0x00007fff2022058d _objc_terminate() + 96
7   libc++abi.dylib               	0x00007fff20335307 std::__terminate(void (*)()) + 8
8   libc++abi.dylib               	0x00007fff20337beb __cxxabiv1::failed_throw(__cxxabiv1::__cxa_exception*) + 27
9   libc++abi.dylib               	0x00007fff20337bb2 __cxa_throw + 116
10  libobjc.A.dylib               	0x00007fff2021dec0 objc_exception_throw + 350
11  com.apple.CoreFoundation      	0x00007fff2050df36 -[NSException raise] + 9
12  com.apple.AppKit              	0x00007fff22c9beb2 -[NSWindow(NSWindow_Theme) _postWindowNeedsToResetDragMarginsUnlessPostingDisabled] + 321
13  com.apple.AppKit              	0x00007fff22c86aa2 -[NSWindow _initContent:styleMask:backing:defer:contentView:] + 1296
14  com.apple.AppKit              	0x00007fff22e48c2f -[NSPanel _initContent:styleMask:backing:defer:contentView:] + 50
15  com.apple.AppKit              	0x00007fff22c8658b -[NSWindow initWithContentRect:styleMask:backing:defer:] + 42
16  com.apple.AppKit              	0x00007fff22e48be4 -[NSPanel initWithContentRect:styleMask:backing:defer:] + 64
17  com.apple.AppKit              	0x00007fff23694274 -[NSSavePanel initWithContentRect:styleMask:backing:defer:] + 97
18  com.apple.AppKit              	0x00007fff2369c36d -[NSOpenPanel initWithContentRect:styleMask:backing:defer:] + 151
19  com.apple.AppKit              	0x00007fff22f950e5 -[NSPanel init] + 75
20  com.apple.AppKit              	0x00007fff236941f0 -[NSSavePanel init] + 197
21  com.apple.AppKit              	0x00007fff23368c3a +[NSSavePanel(Instantiation) _crunchyRawUnbonedPanel] + 58
22  libswnfd.dylib                	0x00000001bf764f56 NFD_OpenDialog + 118
23  ???                           	0x00000001174e2188 0 + 4685963656
24  ???                           	0x00000001174e1cb7 0 + 4685962423
25  ???                           	0x00000001174e1c0d 0 + 4685962253
26  ???                           	0x00000001174e1b87 0 + 4685962119
27  ???                           	0x000000011599f3da 0 + 4657378266
28  ???                           	0x000000011599f342 0 + 4657378114
29  ???                           	0x0000000115992ec2 0 + 4657327810
30  ???                           	0x0000000115992e02 0 + 4657327618
31  ???                           	0x0000000115992395 0 + 4657324949
32  ???                           	0x0000000113d40dd7 0 + 4627631575
33  ???                           	0x0000000113d40b09 0 + 4627630857
34  ???                           	0x0000000113d40a38 0 + 4627630648
35  ???                           	0x0000000111bf7ece 0 + 4592729806
36  libcoreclr.dylib              	0x000000010b867819 CallDescrWorkerInternal + 124
37  libcoreclr.dylib              	0x000000010b6b22cf DispatchCallSimple(unsigned long*, unsigned int, unsigned long, unsigned int) + 239
...
511 libcoreclr.dylib              	0x000000010b6b22cf DispatchCallSimple(unsigned long*, unsigned int, unsigned long, unsigned int) + 239
```
</details>

Full log: [log.txt](https://github.com/space-wizards/RobustToolbox/files/8309006/log.txt)

**Comment by vulppine** [comment]
#2619 should fix this.

**Comment by PJB3005** [comment]
Fixed by c9ccd0b873933a4e84bf8d8e9ad031f65e7d8398 but also credits to https://github.com/space-wizards/RobustToolbox/pull/2619 for doing it before I realized.
